Mathlib/Data/Real/CauSeq.lean

```diff
@@ -139,17 +139,17 @@ section Ring
 
 variable [Ring β] {abv : β → α}
 
-instance : CoeFun (CauSeq β abv) fun _ => ℕ → β :=
-  ⟨Subtype.val⟩
+instance : FunLike (CauSeq β abv) ℕ fun _ => β where
+  coe := Subtype.val
+  coe_injective' := Subtype.val_injective
```

Comment:
I'm not convinced this is a good idea, as now we have two different spellings of `Subtype.val`. I think if you want to use `FunLike`, then probably `CauSeq` should be made a structure rather than a `Subtype`.